### PR TITLE
fix(sequencer): remove failed promotable instead of inserted tx

### PR DESCRIPTION
**Pull request by ethanoroshiba**
## Summary
Fixes an error in mempool promotion logic after insertion.

## Background
Previously, any transactions which failed to be added to the `Pending` queue during promotion post-insertion would remove the just-inserted transaction ID from `contained_txs`. This is an error, but it hasn't caused any problems since this code path is meant to never happen. Additionally, any transactions which failed adding to pending should be added to the CometBFT removal cache.

## Changes
- In `Mempool::insert`, post-insertion, in failure case of adding promoted transaction to `pending`: remove `tx_id` from `contained_txs` instead of the transaction which was successfully added.
- Add failing transaction to CometBFT removal cache.
- Added `RemovalReason::InternalError`.

## Testing
Added unit test to ensure this functionality works as expected. The test scenario is quite contrived, but without direct manipulation of the `pending` and `parked` caches, we have no other way of creating this edge case.

## Changelogs
Changelog updated.

## Related Issues
closes  #2130

### CODE REVIEW

**Review by Fraser999 — APPROVED** [review]

File: crates/astria-sequencer/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Fixed mempool post-insertion failure case during promotion [#2135](https://github.com/astriaorg/astria/pull/2135).
```
Comment (SuperFluffy) [inline crates/astria-sequencer/CHANGELOG.md:14]: You can also amend the changelog to have the (slightly) clearer PR title.

File: crates/astria-sequencer/src/app/test_utils.rs [hunk 2]
```diff
@@ -119,6 +119,15 @@ impl MockTxBuilder {
         }
     }
 
+    pub(crate) fn new_with_chain_id(chain_id: String) -> Self {
```
[the remark this answers is not in the document]
Comment (ethanoroshiba) [reply crates/astria-sequencer/src/app/test_utils.rs]: Removed in 20c6409

File: crates/astria-sequencer/CHANGELOG.md (hunk 1 above)
Comment (ethanoroshiba) [reply crates/astria-sequencer/CHANGELOG.md]: Done in 20c6409.

**Review by noot — APPROVED** [review]
